docs/Advanced/Sync.md

```diff
@@ -170,6 +168,20 @@ Synchronization is serious business! It's very easy to make mistakes that will c
     - … create a record that does exist, update it
   - If there's something wrong with the data format, prefer to "fix" the data if possible instead of failing sync. You don't want the user to have an unsyncable app because of a mistake caused by a bug 5 versions ago.
   - As with any API, remember to check permissions to create/modify records, make sure you version your API together with local Schema versioning, and all other standard stuff!
+- **Adding logging to your sync**
+  - You can add basic sync logs to the sync process by passing an empty object to `synchronize()`. Sync will then mutate the object, populating it with diagnostic information (start/finish time, resolved conflicts, and more):
+
+    ```js
+    const log = {}
+    await synchronize({
+      database,
+      log,
```

Comment:
Hmm wouldn't it be better if user only passes a flag `log: true` and `synchronize` returns `log` object without mutating?

Reply:
BUSTED! 🚓 

You're right, it would be cleaner — I was being lazy ;P Still… there is an issue. `synchronize()` will throw an error (i.e. reject the returned promise) if sync fails — you won't get a log then. So we could either resolve with a log or catch and re-throw all errors inside synchronize by wrapping them in a SyncError and attach a log there… Then a client would have to also handle both cases…

was supposed to be cleaner but… gets a lot uglier. do you think it's worth it?

Reply:
ugh I guess you're right... On the other hand I think that people interested in logs will also catch sync errors anyway but for now let's leave it as it is.